bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusSymoInverterHandler.java

```diff
@@ -84,65 +84,57 @@ protected Object getValue(String channelId) {
         final String fieldName = fields[0];
 
         if (inverterRealtimeResponse != null) {
+            InverterRealtimeBodyData inverterData = inverterRealtimeResponse.getBody().getData();
             switch (fieldName) {
-                case FroniusBindingConstants.INVERTER_DATA_CHANNEL_DAY_ENERGY:
-                    ValueUnit day = inverterRealtimeResponse.getBody().getData().getDayEnergy();
-                    if (day != null) {
-                        day.setUnit("kWh");
-                    }
-                    return day;
                 case FroniusBindingConstants.INVERTER_DATA_CHANNEL_PAC:
-                    ValueUnit pac = inverterRealtimeResponse.getBody().getData().getPac();
-                    if (pac == null) {
-                        pac = new ValueUnit();
-                        pac.setValue(0);
-                    }
-                    return pac;
-                case FroniusBindingConstants.INVERTER_DATA_CHANNEL_TOTAL:
-                    ValueUnit total = inverterRealtimeResponse.getBody().getData().getTotalEnergy();
-                    if (total != null) {
-                        total.setUnit("MWh");
-                    }
-                    return total;
-                case FroniusBindingConstants.INVERTER_DATA_CHANNEL_YEAR:
-                    ValueUnit year = inverterRealtimeResponse.getBody().getData().getYearEnergy();
-                    if (year != null) {
-                        year.setUnit("MWh");
-                    }
-                    return year;
+                    return getQuantityOrZero(inverterData.getPac(), Units.WATT);
                 case FroniusBindingConstants.INVERTER_DATA_CHANNEL_FAC:
-                    return inverterRealtimeResponse.getBody().getData().getFac();
+                    return inverterData.getFac();
```

Comment:
You define a dimension for the channel: Number:Frequency
So a QuantityType is expected.

Reply:
InverterRealtimeBodyData::getFac() returns a `ValueUnit`, which in FroniusBaseThingHandler::updateChannel() gets converted into QuantityType. 

I understand this may not be immediately apparent, so do you think I should do the conversion here and return a QuantityType?

Reply:
Maybe getValue should even be now updated to return a `State` rather than `Object` ?

Reply:
I've refactored it so getValue returns a State.